src/components/SearchBox.vue

```diff
@@ -38,14 +39,16 @@
     methods: {
       ...mapActions(['search', 'updateSearchKeyword', 'searchReset']),
       handleSearch(keyword) {
-        if (keyword === this.$route.query.keyword) {
-          // when keyword is already the same as in url query
-          this.search(keyword)
-        } else {
-          // otherwise update url query
-          this.$router.push({ query: { ...this.$route.query, keyword } })
-          this.search(keyword)
+        clearTimeout(this.debouncedURLChange);
+
+        if (keyword !== this.$route.query.keyword) {
+          // update the url if there wasn't a new search in the last 700ms
+          this.debouncedURLChange = setTimeout(() => {
+            this.$router.push({ query: { ...this.$route.query, keyword } });
+          }, 700);
         }
+
+        this.search(keyword);
```

Comment:
Never mind, I just remove it. https://github.com/egoist/docute/pull/99/commits/5a4a9646ffe55d78a04f9e438df13649e60f220d